### PR TITLE
Add additional controls to 'Other Compliance Checks'  closes #193

### LINKED ISSUE #193: Add additional controls to 'Other Compliance Checks'
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [x] apimanagement_service_client_certificate_enabled
- [x] app_configuration_encryption_enabled
- [x] app_configuration_restrict_public_access
- [x] app_configuration_sku_standard
- [x] appservice_web_app_failed_request_tracing_enabled
- [x] appservice_web_app_http_logs_enabled
- [x] appservice_web_app_latest_dotnet_framework_version
- [x] appservice_web_app_slot_use_https
- [x] appservice_web_app_worker_more_than_one
- [x] container_registry_admin_user_disabled
- [x] container_registry_geo_replication_enabled
- [x] container_registry_public_network_access_disabled
- [x] container_registry_quarantine_policy_enabled
- [x] container_registry_retention_policy_enabled
- [x] container_registry_trust_policy_enabled
- [x] cosmosdb_account_key_based_metadata_write_access_disabled
- [x] data_factory_public_network_access_disabled
- [x] data_factory_uses_git_repository
- [x] eventgrid_domain_identity_provider_enabled
- [x] eventgrid_domain_restrict_public_access
- [x] kubernetes_cluster_addon_azure_policy_enabled
- [x] kubernetes_cluster_node_restrict_public_access
- [x] kubernetes_cluster_restrict_public_access
- [x] kubernetes_cluster_sku_standard
- [x] kubernetes_cluster_upgrade_channel

### CODE REVIEW

File: regulatory_compliance/appconfiguration.sp [hunk 1]
```diff
@@ -40,4 +70,64 @@ query "app_configuration_private_link_used" {
...
       azure_subscription as sub;
   EOQ
+}
+
+query "app_configuration_restrict_public_access" {
+  sql = <<-EOQ
+    select
+      a.id as resource,
+      case
+        when public_network_access = 'Enabled' then 'alarm'
```
Comment (rajlearner17) [inline regulatory_compliance/appconfiguration.sp:80]: @khushboo9024 I guess we have to add a check for private-end-point along with the `public_network_access` attribute?

image

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Review by madhushreeray30 — COMMENTED** [review]
@khushboo9024 please take a look at the comments, thanks!

**Review by misraved — APPROVED** [review]

File: regulatory_compliance/appservice.sp [hunk 2]
```diff
@@ -329,6 +329,56 @@ control "appservice_web_app_latest_python_version" {
...
+  query       = query.appservice_web_app_http_logs_enabled
+
+  tags = merge(local.regulatory_compliance_appservice_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "appservice_web_app_worker_more_than_one" {
+  title       = "Web app should have more than one worker"
+  description = "It is recommended to have more than one worker for failover. This control is non-compliant if Web app have one or less than one worker."
```
Comment (misraved) [inline regulatory_compliance/appservice.sp:364]:
```suggestion
  description = "It is recommended to have more than one worker for failover. This control is non-compliant if Web apps have one or less than one worker."
```

File: regulatory_compliance/eventgrid.sp [hunk 3]
```diff
@@ -24,6 +24,26 @@ control "eventgrid_topic_private_link_used" {
   })
 }
 
+control "eventgrid_domain_restrict_public_access" {
+  title       = "Event Grid domains should restrict public network access"
+  description = "Ensure that Event Grid Domain public network access is disabled. This control is non-compliant if Event Grid domain have public network access enabled."
```
Comment (misraved) [inline regulatory_compliance/eventgrid.sp:29]:
```suggestion
  description = "Ensure that Event Grid Domain public network access is disabled. This control is non-compliant if Event Grid domains have public network access enabled."
```

File: regulatory_compliance/eventgrid.sp [hunk 4]
```diff
@@ -24,6 +24,26 @@ control "eventgrid_topic_private_link_used" {
...
+  query       = query.eventgrid_domain_restrict_public_access
+
+  tags = merge(local.regulatory_compliance_eventgrid_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "eventgrid_domain_identity_provider_enabled" {
+  title       = "Event Grid domains identity provider should be enabled"
+  description = "Ensure that managed identity provider is enabled for Event Grid Domain. This control is non-compliant if Event Grid domain identity provides disabled."
```
Comment (misraved) [inline regulatory_compliance/eventgrid.sp:39]:
```suggestion
  description = "Ensure that managed identity provider is enabled for Event Grid Domain. This control is non-compliant if Event Grid domain identity provider is disabled."
```

File: regulatory_compliance/kubernetes.sp [hunk 5]
```diff
@@ -205,6 +205,56 @@ control "kubernetes_cluster_https_enabled" {
...
+  query       = query.kubernetes_cluster_addon_azure_policy_enabled
+
+  tags = merge(local.regulatory_compliance_kubernetes_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "kubernetes_cluster_node_restrict_public_access" {
+  title       = "Kubernetes cluster nodes should prohibit public access"
+  description = "Ensure Kubernetes cluster node do not have public IP addresses. This control is non-compliant if Kubernetes cluster node have public IP address assigned."
```
Comment (misraved) [inline regulatory_compliance/kubernetes.sp:230]:
```suggestion
  description = "Ensure Kubernetes cluster nodes do not have public IP addresses. This control is non-compliant if Kubernetes cluster nodes have a public IP address assigned."
```

**Pull request merged**